### DISCUSSION TITLE
Proposal to Use typer for CLI and Importable Module Development

### DISCUSSION THREAD

**Discussion by porimol** [Bangla-RAG/PoRAG#5] [Ideas]
### Why?

I would like to propose to use typer for developing a command-line interface (CLI) and making `PoRAG` library an importable module. [typer](https://typer.tiangolo.com/tutorial/typer-command/) is a modern, user-friendly library that leverages Python `type hints`,  easy to develop CLIs that are both intuitive and maintainable. This change will not only enhance `PoRAG` capabilities but also allow the functionalities to be easily imported and reused in other Python project.

**Example:**

```python
# cli/hello.py
import typer

app = typer.Typer()

@app.command()
def porag(data_path: str):
    """
    Just yet another docstring...
    """
    typer.echo(f"This is the {data_path} path.")

if __name__ == "__main__":
    app()
```
Import in other project
```bash
from cli import porag

porag("World")
```

Use as `CLI`
```bash
porag --data_path test.txt
```

References:
[Typer Documentation](https://typer.tiangolo.com/tutorial/typer-command/)
[Typer GitHub Repository](https://github.com/tiangolo/typer)

**Comment by asiff00** [comment]
@porimol Thanks! We will explore that!

**Comment by hassanaliemon** [comment]
Hello @porimol ,
our main goal is to make an UI for the whole process and what we have published is just an initial one. Very soon UI will come and everyone will be able to directly chat with our PoRAG like we do in our messenger app. 

But we appreciate your concern and expect to get more feedback from you in future.
